src/phasorpy/components.py

```diff
@@ -85,32 +92,155 @@ def two_fractions_from_phasor(
-        [real_components[1], imag_components[1]]
+    components_real = numpy.asarray(components_real)
+    components_imag = numpy.asarray(components_imag)
+    if components_real.shape != (2,):
+        raise ValueError(f'{components_real.shape=} != (2,)')
+    if components_imag.shape != (2,):
+        raise ValueError(f'{components_imag.shape=} != (2,)')
+    _, distance_between_components = line_from_components(
+        components_real, components_imag
     )
-    total_distance_between_components = math.hypot(
-        (second_component_phasor[0] - first_component_phasor[0]),
-        (second_component_phasor[1] - first_component_phasor[1]),
-    )
-    if math.isclose(total_distance_between_components, 0, abs_tol=1e-6):
-        raise ValueError('components must have different coordinates')
     projected_real, projected_imag = project_phasor_to_line(
-        real, imag, real_components, imag_components
+        real, imag, components_real, components_imag
     )
     distances_to_first_component = numpy.hypot(
-        numpy.asarray(projected_real) - first_component_phasor[0],
-        numpy.asarray(projected_imag) - first_component_phasor[1],
+        numpy.asarray(projected_real) - components_real[0],
+        numpy.asarray(projected_imag) - components_imag[0],
     )
-    fraction_of_second_component = (
-        distances_to_first_component / total_distance_between_components
+    second_component_fractions = (
+        distances_to_first_component / distance_between_components
     )
-    return 1 - fraction_of_second_component, fraction_of_second_component
+    first_component_fractions = 1 - second_component_fractions
+    return first_component_fractions, second_component_fractions
+
+
+def graphical_component_analysis(
+    real: ArrayLike,
+    imag: ArrayLike,
+    components_real: ArrayLike,
+    components_imag: ArrayLike,
+    /,
+    *,
+    radius: float = 0.05,
+    steps: int = 100,
+) -> tuple[tuple[NDArray[Any], ...], NDArray[Any]]:
+    """Return fractions of two or three components from phasor coordinates.
+
+    The graphical method is based on moving circular cursors along the line
+    between pairs of components, and quantifying the phasors for each fraction.
+
+    Parameters
+    ----------
+    real : array_like
+        Real component of phasor coordinates.
+    imag : array_like
+        Imaginary component of phasor coordinates.
+    components_real: array_like
+        Real coordinates for two or three components.
+    components_imag: array_like
+        Imaginary coordinates for two or three components.
+    radius: float, optional
+        Diameter of the cursor in phasor coordinates.
+    steps: int, optional
+        Number of steps to move the cursor along the line between components.
+
+    Returns
+    -------
+    fractions : tuple of ndarray
+        Counts along each line segment connecting the components, ordered
+        1-2, 1-3, 2-3 (for 3 components) or simply 1-2 (for 2 components).
```

Comment:
Python uses zero-based indexing: `0-1, 0-2, 1-2 (for 3 components) or simply 0-1 (for 2 components).`